paging/src/iosMain/kotlin/com/kuuurt/paging/multiplatform/Pager.kt

```diff
@@ -21,16 +21,20 @@ actual class Pager<K : Any, V : Any> actual constructor(
     private val initialKey: K,
     private val getItems: suspend (K, Int) -> PagingResult<K, V>
 ) {
-    private val items = PagingData<V>()
 
-    private val _pagingData = MutableStateFlow<PagingData<V>?>(null)
-    actual val pagingData: Flow<PagingData<V>> get() = _pagingData.filterNotNull()
+    private data class State<K : Any, V : Any>(
```

Comment:
Do we really have to use this `State` class? I feel that we may just be adding extra objects to manage here. Could we just use the variables instead so we can avoid `state.copy`? Or is this essential to workaround the freezing issue?

Reply:
The `State` object isn't really needed for the freezing issue. It could also be accomplished with 3 separate `StateFlow` properties. The `State` object does has an extra benefit of preventing duplicate items in cases where multiple load operations are running in parallel. I am not sure if that's really useful since it will now only contain the items from the second call. Ideally I guess the calls shouldn't run in parallel anyway.

What do you think? I could change it to use 3 separate `StateFlow`s if you like.

Reply:
I think I'd prefer that rather than adding this extra layer. Please make the changes then I'll release the next version as soon as I can 😄